### PR TITLE
Issue 2011 - Soundcloud oembed provider

### PR DESCRIPTION
**Pull request by elcazador**
Modified the Soundcloud provider by adding the concept of a formatter
that can be added to a provider and needed.  The formatter has access to
and can return modified html embed code.

Added global title code for iframe and embed tags for rich media

### CODE REVIEW

File: libs/jquery.oembed.js [hunk 1]
```diff
@@ -73,7 +73,7 @@
                                     provider = $.fn.oembed.getOEmbedProvider(data['long-url']);
 
                                     //remove fallback
-                                    if (!!settings.fallback === false) {
+                                    if (settings.fallback === false) {
```
Comment (DaveBagler) [inline libs/jquery.oembed.js:76]: Do we get type coercion issues on this if statement?

File: libs/jquery.oembed.js [hunk 2]
```diff
@@ -101,7 +101,7 @@
                 provider = $.fn.oembed.getOEmbedProvider(resourceURL);
 
                 //remove fallback
-                if (!!settings.fallback === false) {
+                if (settings.fallback === false) {
```
Comment (DaveBagler) [inline libs/jquery.oembed.js:104]: Same question about types here.

File: libs/jquery.oembed.js [hunk 3]
```diff
@@ -762,7 +784,17 @@
         new $.fn.oembed.OEmbedProvider("shoudio", "rich", ["shoudio.com/.+", "shoud.io/.+"], "http://shoudio.com/api/oembed"),
         new $.fn.oembed.OEmbedProvider("mixcloud", "rich", ["mixcloud.com/.+"], 'http://www.mixcloud.com/oembed/', {useYQL: 'json'}),
         new $.fn.oembed.OEmbedProvider("rdio.com", "rich", ["rd.io/.+", "rdio.com"], "http://www.rdio.com/api/oembed/"),
-        new $.fn.oembed.OEmbedProvider("Soundcloud", "rich", ["soundcloud.com/.+", "snd.sc/.+"], "//soundcloud.com/oembed", {format: 'js'}),
+        new $.fn.oembed.OEmbedProvider("Soundcloud", "rich", ["soundcloud.com/.+", "snd.sc/.+"], "//soundcloud.com/oembed", {
```
Comment (DaveBagler) [inline libs/jquery.oembed.js:787]: Yeah, that's way easier to read now. 👍

File: libs/jquery.oembed.js (hunk 1 above)
Comment (elcazador) [reply libs/jquery.oembed.js:76]: Shouldn't be a problem unless someone is being a dingus.  Default values are sensible.

**Review by walkerjam — APPROVED** [review]

**Review by DaveBagler — APPROVED** [review]

**Review by budimanc — APPROVED** [review]
Wow...haven't seen double-bang operator in a while.

**Pull request merged**